File: simulators/de.unistuttgart.iste.sqa.mpw.hamstersimulator.java/examples/src/main/java/de/unistuttgart/hamster/examples/Example03.java
package de.unistuttgart.hamster.examples;

import de.unistuttgart.hamster.main.SimpleHamsterGame;

public class Example03 extends SimpleHamsterGame {
    public static void main(final String[] args) {
        createInstance(Example03.class);
    }

    /**
     * Another hamster program, which tests the loading of another territory.
     */
    @Override
    protected void run() {
        displayInNewGameWindow();
        loadTerritoryFromResourceFile("de.unistuttgart.hamster.territories/example03.ter");
        startGame();

        turnRight();
        paule.move();
    }

    void turnRight() {
        paule.turnLeft();
        paule.turnLeft();
        paule.turnLeft();
    }
}

File: examples/Example03.cpp
#include "SimpleHamsterGame.h"

using namespace hamster;

class Example03 : public hamstersimulator::SimpleHamsterGame {
protected:
    void run() override;
    void turnRight();
};

void Example03::run() {
    loadTerritoryFromResourceFile("resources/territories/example03.ter");
    startGame();

    turnRight();
    paule.move();
}

void Example03::turnRight() {
    paule.turnLeft();
    paule.turnLeft();
    paule.turnLeft();
}

int main() {
    hamstersimulator::SimpleHamsterGame::createInstance<Example03>();
    return 0;
}
